src/timeUtils.ts

```diff
@@ -1148,14 +1191,17 @@ export function parseDOYDurationTime(
       const minuteNum = parseInt(mins);
       const secondsNum = parseInt(secs);
       const millisecondNum = parseFloat(
-        (parseFloat(`.${ms}`) * 1000).toFixed(6),
+        (parseFloat(`.${ms.substring(0, 3)}`) * 1000).toFixed(6)
+      );
+      const microsecondsNum = parseFloat(
+        (parseFloat(`.${ms.substring(3)}`) * 1000).toFixed(6)
```

Comment:
Review this function

Reply:
I have updated the regex and pushed commits to support <us> group

2ac5bad930c3c66fe08fe0b999ceef4fb8795173